### PR TITLE
Validate OAuth2 redirect uri

### PR DESCRIPTION
**Pull request by rullzer**
Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: apps/oauth2/src/App.vue [hunk 1]
```diff
@@ -92,21 +95,28 @@ export default {
...
+				console.log(response);
+				this.clients.push(response.data)
 
-					this.newClient.name = '';
-					this.newClient.redirectUri = '';
-				}
-			);
+				this.newClient.name = '';
+				this.newClient.redirectUri = '';
+			}).catch((reason) => {
```
Comment (skjnldsv) [inline apps/oauth2/src/App.vue]: actually you can just to like `.catch(reason => {`
Not a big deal, just as a record :p

File: apps/oauth2/src/App.vue [hunk 2]
```diff
@@ -92,21 +95,28 @@ export default {
...
 				{
 					name: this.newClient.name,
 					redirectUri: this.newClient.redirectUri
 				},
-				tokenHeaders)
-				.then((response) => {
-					this.clients.push(response.data)
+				tokenHeaders
+			).then((response) => {
+				console.log(response);
```
Comment (skjnldsv) [inline apps/oauth2/src/App.vue]: old debug?
If so you can also  do `.then(response => this.clients.push(response.data)).catch`

File: apps/oauth2/src/App.vue (hunk 1 above)
Comment (rullzer) [reply apps/oauth2/src/App.vue]: right :)

File: apps/oauth2/src/App.vue (hunk 2 above)
Comment (rullzer) [reply apps/oauth2/src/App.vue]: yep fixed

**Review by skjnldsv — APPROVED** [review]

File: apps/oauth2/src/App.vue [hunk 3]
```diff
@@ -92,21 +95,24 @@ export default {
...
+				this.clients.push(response.data);
 
-					this.newClient.name = '';
-					this.newClient.redirectUri = '';
-				}
-			);
+				this.newClient.name = '';
+				this.newClient.redirectUri = '';
+			}).catch(reason => {
+				this.newClient.error = true;
```
Comment (skjnldsv) [inline apps/oauth2/src/App.vue]: you could have merged the two variables into one. If it's a string and not false, then it's most definitely an error ;)

**Review by ChristophWurst — APPROVED** [review]
LGTM!

File: apps/oauth2/src/App.vue (hunk 3 above)
Comment (rullzer) [reply apps/oauth2/src/App.vue]: future work :P

**Comment by MorrisJobke** [comment]
CI fails hard

**Comment by rullzer** [comment]
Should be good now

**Comment by skjnldsv** [comment]
Failure unrelated

**Pull request merged**